tests/bluetooth/tester/src/btp_micp.c

```diff
@@ -0,0 +1,534 @@
+}
+
+static void micp_mic_ctlr_aics_mute_cb(struct bt_aics *inst, int err)
+{
+	LOG_DBG("AICS mute cb (%d)", err);
+}
+
+static void micp_mic_ctlr_aics_set_manual_mode_cb(struct bt_aics *inst, int err)
+{
+	LOG_DBG("AICS Set manual gain mode cb (%d)", err);
+}
+
+static void micp_mic_ctlr_aics_automatic_mode_cb(struct bt_aics *inst, int err)
+{
+	LOG_DBG("AICS set automatic gain mode cb (%d)", err);
+}
+
+static struct bt_micp_mic_ctlr_cb micp_cbs = {
+	.discover = micp_mic_ctlr_discover_cb,
+	.mute = micp_mic_ctlr_mute_cb,
+	.mute_written = micp_mic_ctlr_mute_written_cb,
+	.unmute_written = micp_mic_ctlr_unmute_written_cb,
+
+
+#if defined(CONFIG_BT_MICP_MIC_CTLR_AICS)
+	/* Audio Input Control Service */
+	.aics_cb = {
+		.state = micp_mic_ctlr_aics_state_cb,
+		.gain_setting = micp_mic_ctlr_aics_gain_setting_cb,
+		.type = micp_mic_ctlr_aics_input_type_cb,
+		.status = micp_mic_ctlr_aics_status_cb,
+		.description = micp_mic_ctlr_aics_description_cb,
+		.set_gain = micp_mic_ctlr_aics_set_gain_cb,
+		.unmute = micp_mic_ctlr_aics_unmute_cb,
+		.mute = micp_mic_ctlr_aics_mute_cb,
+		.set_manual_mode = micp_mic_ctlr_aics_set_manual_mode_cb,
+		.set_auto_mode = micp_mic_ctlr_aics_automatic_mode_cb,
+	}
+#endif /* CONFIG_BT_MICP_MIC_CTLR_AICS */
+};
+
+static uint8_t micp_supported_commands(const void *cmd, uint16_t cmd_len,
+				       void *rsp, uint16_t *rsp_len)
+{
+	struct btp_micp_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_MICP_READ_SUPPORTED_COMMANDS);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_discover(const void *cmd, uint16_t cmd_len,
+			     void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_micp_discover_cmd *cp = cmd;
+	struct bt_conn *conn;
+	int err;
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+		return BTP_STATUS_FAILED;
+	}
+
+	err = bt_micp_mic_ctlr_discover(conn, &mic_ctlr);
+
+	if (err != 0) {
+		LOG_DBG("Fail: %d", err);
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_set_gain(const void *cmd, uint16_t cmd_len,
+			     void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_micp_set_gain_cmd *cp = cmd;
+
+	LOG_DBG("MICP set gain %d", cp->gain);
+
+	for (int i = 0; i < micp_included.aics_cnt; i++) {
+		if (bt_aics_gain_set(micp_included.aics[i], cp->gain) != 0) {
+			return BTP_STATUS_FAILED;
+		}
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_aics_unmute(const void *cmd, uint16_t cmd_len,
+				void *rsp, uint16_t *rsp_len)
+{
+	LOG_DBG("Unmute");
+
+	for (int i = 0; i < micp_included.aics_cnt; i++) {
+		if (bt_aics_unmute(micp_included.aics[i]) != 0) {
+			return BTP_STATUS_FAILED;
+		}
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_aics_mute(const void *cmd, uint16_t cmd_len,
+			      void *rsp, uint16_t *rsp_len)
+{
+	LOG_DBG("Mute");
+
+	for (int i = 0; i < micp_included.aics_cnt; i++) {
```

Comment:
```suggestion
	for (uint8_t i = 0; i < micp_included.aics_cnt; i++) {
```

Applies more places too